docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/FeatureSelection/SelectFeaturesBasedOnCount.cs

```diff
@@ -0,0 +1,112 @@
+using Microsoft.ML;
+using Microsoft.ML.Data;
+
+namespace Samples.Dynamic
+{
+    public static class SelectFeaturesBasedOnCount
+    {
+        public static void Example()
+        {
+            // Create a new ML context, for ML.NET operations. It can be used for exception tracking and logging, 
+            // as well as the source of randomness.
+            var mlContext = new MLContext();
+
+            // Get a small dataset as an IEnumerable and convert it to an IDataView.
+            var rawData = GetData();
+            var data = mlContext.Data.LoadFromEnumerable(rawData);
+
+            var convertedData = mlContext.Data.CreateEnumerable<InputData>(data, true);
+
+            Console.WriteLine("Contents of two columns 'NumericVector' and 'StringVector'.");
+            foreach (var item in convertedData)
+                Console.WriteLine("{0}\t\t\t{1}", string.Join("\t", item.NumericVector), string.Join("\t", item.StringVector));
+            // 4       NaN     6                       A       WA   Male
+            // 4       5       6                       A            Female
+            // 4       5       6                       A       NY
+            // 4       NaN     NaN                     A            Male
+
+            // We will use the SelectFeaturesBasedOnCount to retain only those slots which have at least 'count' non-default values per slot.
+
+            // Usage on numeric column.
+            var pipeline = mlContext.Transforms.FeatureSelection.SelectFeaturesBasedOnCount(
+                outputColumnName: "NumericVector", count: 3);
+
+            // The pipeline can then be trained, using .Fit(), and the resulting transformer can be used to transform data. 
+            var transformedData = pipeline.Fit(data).Transform(data);
+
+            Console.WriteLine("Contents of column 'NumericVector'");
+            var featuresSelectedGroupA = transformedData.GetColumn<float[]>(transformedData.Schema["NumericVector"]);
+            foreach (var row in featuresSelectedGroupA)
+            {
+                for (var i = 0; i < row.Length; i++)
+                    Console.Write($"{row[i]}\t");
+                Console.WriteLine();
+            }
+            // 4       6
+            // 4       6
+            // 4       6
+            // 4       NaN
+
+            // Usage on text column.
+            pipeline = mlContext.Transforms.FeatureSelection.SelectFeaturesBasedOnCount(
+                outputColumnName: "StringVector", count: 3);
+
+            transformedData = pipeline.Fit(data).Transform(data);
+
+            Console.WriteLine("Contents of column 'StringVector'");
+            var featuresSelectedInfo = transformedData.GetColumn<string[]>(transformedData.Schema["StringVector"]);
+            foreach (var row in featuresSelectedInfo)
+            {
+                for (var i = 0; i < row.Length; i++)
+                    Console.Write($"{row[i]}\t");
+                Console.WriteLine();
+            }
```

Comment:
would it polish it a bit if you made a little helper for this, since it is being used twice?

Reply:
one of then is a float[]  while the other is string[] ..  did not want to over-engineer this...

is there anything specific you had in mind, or can we keep it as is ?

---
In reply to: [272392068](https://github.com/dotnet/machinelearning/pull/3184#discussion_r272392068) [](ancestors = 272392068)

Reply:
object :)

---
In reply to: [272394600](https://github.com/dotnet/machinelearning/pull/3184#discussion_r272394600) [](ancestors = 272394600,272392068)